pipeline/export/nwb.py

```diff
@@ -42,93 +48,293 @@ def datajoint_to_nwb(session_key):
+
+    # ---- photostim events ----
+    q_photostim_event = (experiment.TrialEvent * experiment.PhotostimEvent & session_key).proj(
+        event_start='trial_event_time', event_stop = '(trial_event_time+duration)', 
+        power= 'power', photostim='photo_stim')
+
+    if q_photostim_event:
+        trial_event_id, event_starts, event_stops, powers, photo_stim = q_photostim_event.fetch(
+            'trial_event_id','event_start', 'event_stop', 'power', 'photostim', order_by='trial')
+        
+        all_times = [event_starts[0]]
+
+        for time in all_session_times.fetch('duration'):
+            all_times.append(time)
+            all_times_arr = np.cumsum(all_times)
+
+        all_times_arr = np.delete(all_times_arr, -1)
+        trial_start_times = []
+
+        for idx, times in enumerate(all_times_arr):
+            for trial in trial_event_id:
+                if trial == idx:
+                    trial_start_times.append(times)
+
+                
+        behav_event.create_timeseries(name='photostim_start_times', unit='mW', conversion=1.0,
+                                    description='Timestamps of the photo-stimulation and the corresponding powers (in mW) being applied',
+                                    data=powers.astype(float),
+                                    timestamps=trial_start_times,
+                                    control=photo_stim.astype('uint8'), control_description=stim_sites)
+        behav_event.create_timeseries(name='photostim_stop_times', unit='mW', conversion=1.0,
+                                    description = 'Timestamps of the photo-stimulation being switched off',
+                                    data=np.full_like(event_starts.astype(float), 0),
+                                    timestamps=event_stops.astype(Decimal) + trial_start_times,
+                                    control=photo_stim.astype('uint8'), control_description=stim_sites)
+
+    # ---- behavior events ----
+
+    q_behavior_trials = ((experiment.TrialEvent * experiment.BehaviorTrial) - experiment.PhotostimEvent 
+                        & session_key).proj(event_start='trial_event_time', event_stop = '(trial_event_time+duration)',
+                            outcome='outcome',trial_instruction='trial_instruction')
+
+    if q_behavior_trials:
+        trial_event_id, event_starts, event_stops, powers, photo_stim = q_behavior_trials.fetch(
+            'trial_event_id','event_start', 'event_stop', 'outcome', 'trial_instruction', order_by='trial')
+ 
+        all_times = [event_starts[0]]
+
+        for time in all_session_times.fetch('duration'):
+            all_times.append(time)
+            all_times_arr = np.cumsum(all_times)
+
+        all_times_arr = np.delete(all_times_arr, -1)
+
+        trial_start_times = []
+
+        for idx, times in enumerate(all_times_arr):
+            for trial in trial_event_id:
+                if trial == idx:
+                    trial_start_times.append(times)
```

Comment:
There should be a much better / clearer way of doing this.

Reply:
This ended up being removed.